code/datums/mind.dm

```diff
@@ -1562,14 +1562,28 @@
 /datum/mind/proc/find_syndicate_uplink()
 	var/list/L = current.get_contents()
 	for(var/obj/item/I in L)
-		if(I.hidden_uplink)
+		if(I.hidden_uplink) //find items disguised as uplinks
 			return I.hidden_uplink
-	return null
+		if(istype(I, /obj/item/implant/uplink)) //find uplink implants
+			return I
+	return
 
 /datum/mind/proc/take_uplink()
-	var/obj/item/uplink/hidden/H = find_syndicate_uplink()
+	var/list/A = current.actions //remove uplink implant action button
+	for(var/datum/action/item_action/hands_free/activate/I in A)
+		var/obj/item/implant/uplink/U = I.target
+		if(!isnull(U.hidden_uplink))
```

Comment:
Question, `hidden_uplink` will never hold a `0` integer or blank string correct? If so, we don't need to perform an `isnull` check here and can just simplify this to
```dm
if(!U.hidden_uplink)
```